app/Models/Api/Search.php

```diff
@@ -512,19 +512,16 @@ public function scopeByColor($query, $hsl)
         return $query->rawSearch($params);
     }
 
-    public function scopeByMostSimilar($query, $id, $class = \App\Models\Api\Artwork::class, $boost = false)
+    public function scopeByMostSimilar($query, Artist|Artwork $item = null, $boost = false)
     {
-        if (empty($id)) {
+        if (empty($item)) {
             return $query;
         }
+        $class = get_class($item);
```

Comment:
This refactor prevents a duplicate API request. Instead of retrieving an ApiModel, passing the id and class to this scope, and then retrieving the ApiModel again, we can just pass the ApiModel directly to the scope.